enhancements/hypershift/priority-class/priority-class.md

```diff
@@ -0,0 +1,266 @@
+  - apiVersion: scheduling.k8s.io/v1
+    kind: PriorityClass
+    metadata:
+      name: hypershift-etcd-m
+    value: 3500000
+    globalDefault: false
+    description: "Used on master ETCD resources for cruisers with 26-50 workers."
+  - apiVersion: scheduling.k8s.io/v1
+    kind: PriorityClass
+    metadata:
+      name: hypershift-l
+    value: 4000000
+    globalDefault: false
+    description: "Used on master control plane resources for cruisers with 51-100 workers."
+  - apiVersion: scheduling.k8s.io/v1
+    kind: PriorityClass
+    metadata:
+      name: hypershift-etcd-l
+    value: 4500000
+    globalDefault: false
+    description: "Used on master ETCD resources for cruisers with 51-100 workers."
+  - apiVersion: scheduling.k8s.io/v1
+    kind: PriorityClass
+    metadata:
+      name: hypershift-xl
+    value: 5000000
+    globalDefault: false
+    description: "Used on master control plane resources for cruisers with 101-300 workers."
+  - apiVersion: scheduling.k8s.io/v1
+    kind: PriorityClass
+    metadata:
+      name: hypershift-etcd-xl
+    value: 5500000
+    globalDefault: false
+    description: "Used on master ETCD resources for cruisers with 101-300 workers."
+  - apiVersion: scheduling.k8s.io/v1
+    kind: PriorityClass
+    metadata:
+      name: hypershift-xxl
+    value: 6000000
+    globalDefault: false
+    description: "Used on master control plane resources for cruisers with 300+ workers."
+  - apiVersion: scheduling.k8s.io/v1
+    kind: PriorityClass
+    metadata:
+      name: hypershift-etcd-xxl
+    value: 6500000
+    globalDefault: false
+    description: "Used on master ETCD resources for cruisers with 300+ workers."
+  - apiVersion: scheduling.k8s.io/v1
+    kind: PriorityClass
+    metadata:
+      name: hypershift-dedicated
+    value: 7000000
+    globalDefault: false
+    description: "Used on master control plane resources for dedicated cruisers."
+  - apiVersion: scheduling.k8s.io/v1
+    kind: PriorityClass
+    metadata:
+      name: hypershift-etcd-dedicated
+    value: 7500000
+    globalDefault: false
+    description: "Used on master ETCD resources for dedicated cruisers."
+```
+
+With the availablity of the above priority classes, HO can be enhanced to use and override priority classes being on set on the underlying control plane workloads/pods.
+
+### Workflow Description
+TBD
+### API Extensions
+HC and HCP CRDs -- new field
+
+### Implementation Details/Notes/Constraints
+This effort will be implemented over two phases:
+- Phase 1 - Override Priority Classes
+- Phase 2 - Dynamic Adjustment of Priority Classes
+
+### Phase 1 - Override Priority Classes
+Hypershift currently creates and uses a static set of default priority classes. The first phase of this effort will largely focus on Hypershift's ability to override these priority classes for underlying workloads.
+- The overridden priority classes must be specified in the newly created API fields for the HC. 
```

Comment:
It would be good to describe what the API change would look like.

Reply:
An alternative to consider for an initial implementation would be to allow specifying priority class names as annotations in the HostedCluster while we work out what the best API for it should look like.

Reply:
Will change our approach here to show we intend to use the annotations in the HostedCluster instead.